integrations/ollama/src/haystack_integrations/components/embedders/ollama/document_embedder.py

```diff
@@ -160,7 +160,7 @@ async def _embed_batch_async(
         return all_embeddings
 
     @component.output_types(documents=List[Document], meta=Dict[str, Any])
-    def run(self, documents: List[Document], generation_kwargs: Optional[Dict[str, Any]] = None):
+    def run(self, documents: List[Document], generation_kwargs: Optional[Dict[str, Any]] = None) -> Union[Dict[str, List[Document]], Dict[str, Dict[str, Any]]]:
```

Comment:
Wait, shouldn't return type be Dict[str, Union[List[Document], Dict[str, Any]]]?

Reply:
right. Done in https://github.com/deepset-ai/haystack-core-integrations/pull/1922/commits/b9bc2154d229ce7533b687861f0dd5f71199d950